sklearn/decomposition/factor_analysis.py

```diff
@@ -228,6 +238,8 @@ def my_svd(X):
                           ConvergenceWarning)
 
         self.components_ = W
+        if self.rotation is not None:
```

Comment:
might as well do this check inside the method

Reply:
Actually, replace the previous line to `self.components_ = self._rotate(W)`. I think that would be much clearer to the reader.